erizo/src/erizo/rtp/PliPriorityHandler.cpp

```diff
@@ -0,0 +1,86 @@
+#include "rtp/PliPriorityHandler.h"
+
+#include "rtp/RtpUtils.h"
+#include "./MediaDefinitions.h"
+#include "./MediaStream.h"
+
+namespace erizo {
+
+DEFINE_LOGGER(PliPriorityHandler, "rtp.PliPriorityHandler");
+
+constexpr duration PliPriorityHandler::kLowPriorityPliPeriod;
+
+PliPriorityHandler::PliPriorityHandler(std::shared_ptr<erizo::Clock> the_clock)
+    : enabled_{true}, stream_{nullptr}, clock_{the_clock},
+      video_sink_ssrc_{0}, video_source_ssrc_{0}, plis_received_in_interval_{0}, first_received_{false} {}
+
+void PliPriorityHandler::enable() {
+  enabled_ = true;
+}
+
+void PliPriorityHandler::disable() {
+  enabled_ = false;
+}
+
+void PliPriorityHandler::notifyUpdate() {
+  auto pipeline = getContext()->getPipelineShared();
+  if (pipeline && !stream_) {
+    stream_ = pipeline->getService<MediaStream>().get();
+    video_sink_ssrc_ = stream_->getVideoSinkSSRC();
+    video_source_ssrc_ = stream_->getVideoSourceSSRC();
+  }
+}
+
+void PliPriorityHandler::read(Context *ctx, std::shared_ptr<DataPacket> packet) {
```

Comment:
shouldn't we check if we are sending keyframes to reset `plis_received_in_interval_` to 0?